data_structures/skip_list.cpp

```diff
@@ -0,0 +1,212 @@
+    int lvl = 0; 
+    while(static_cast<float>(std::rand())/RAND_MAX < P && lvl < MAXLVL) lvl++;  
+    return lvl; 
+} 
+  
+  
+// Insert given key in skip list 
+void SkipList::insertElement(int key, void* value) { 
+    std::cout << "Inserting" << key << "...";
+    Node *x = header; 
+    Node *update[MAXLVL+1]; 
+    memset(update, 0, sizeof(Node*)*(MAXLVL+1)); 
+
+
+    for(int i = level; i >= 0; i--) { 
+        while(x->forward[i] != NULL && x->forward[i]->key < key)  x = x->forward[i]; 
+        update[i] = x; 
+    } 
+
+    x = x->forward[0]; 
+
+    bool doesnt_exist = (x == NULL || x->key != key);
+    if (doesnt_exist) { 
+        int rlevel = randomLevel(); 
+  
+        if(rlevel > level) { 
+            for(int i=level+1;i<rlevel+1;i++) update[i] = header; 
+
+            /*Update current level*/
+            level = rlevel; 
+        } 
+  
+        Node* n = new Node(key, rlevel, value);
+        for(int i=0;i<=rlevel;i++) { 
+            n->forward[i] = update[i]->forward[i]; 
+            update[i]->forward[i] = n; 
+        } 
+        std::cout << "Inserted" << endl;
+
+    } else {
+        std::cout << "Exists" << endl;
+    }
+} 
+  
+  /**Delete document by key*/
+void SkipList::deleteElement(int key) 
+{ 
+    Node *x = header; 
+  
+    Node *update[MAXLVL+1]; 
+    memset(update, 0, sizeof(Node*)*(MAXLVL+1)); 
+  
+    for(int i = level; i >= 0; i--)  { 
+        while(x->forward[i] != NULL  &&  x->forward[i]->key < key)  x = x->forward[i]; 
+        update[i] = x; 
+    } 
+  
+    x = x->forward[0]; 
+  
+    bool doesnt_exist = (x == NULL || x->key != key);
+
+    if(!doesnt_exist) { 
+        for(int i=0;i<=level;i++) {
+            if(update[i]->forward[i] != x)  break; 
+            update[i]->forward[i] = x->forward[i]; 
+        }
+       /*Remove empty levels*/
+        while(level>0 &&  header->forward[level] == 0) level--; 
+        std::cout << "Deleted" << endl;
+    } else {
+        std::cout << "Doesn't exist" << endl;
+    }
+} 
+  
+
+/**
+ * Searching element in skip list structure
+*/
+void* SkipList::searchElement(int key) { 
+    Node *x = header; 
+    std::cout << "Searching for " + key << endl;
+
+    for(int i = level; i >= 0; i--) { 
+        while(x->forward[i] && x->forward[i]->key < key)  x = x->forward[i]; 
+    } 
+  
+    x = x->forward[0]; 
+    if(x && x->key == key){
+        std::cout << "Found" << endl;
+        return x->value;
+    } else {
+        std::cout << "Not Found" << endl;
+        return NULL;
+    }
+} 
+  
+// Display skip list level wise 
+void SkipList::displayList() { 
+    std::cout << "Displaying list:\n" << endl;
+    for(int i=0; i <= level; i++) { 
+        Node *node = header->forward[i]; 
+        std::cout << "Level " << (i) << ": ";
+        while(node != NULL) { 
+            std::cout << node->key << " ";
+            node = node->forward[i]; 
+        } 
+        std::cout << endl;
+    } 
+} 
+  
+int main() 
+{ 
+    std::srand(time(nullptr));
```

Comment:
```suggestion
    std::srand(std::time(nullptr));
```

Reply:
I tried that but compliator tells me that "error: ‘time’ is not a member of ‘std’"

Reply:
thats because, you have been using the C libraries and not the C++

Reply:
Ok, changed into <ctime>, thank you again